includes/ets_user_qa_function.php

```diff
@@ -195,176 +197,193 @@ public function ets_ask_qustion_tab() {
 				);
 			?>
 			<?php  
-			}  
-			$loadMoreButtonName = get_option('ets_load_more_button_name');
-			$productQaLength = get_option('ets_product_q_qa_list_length'); 
-			$loadMoreButton = get_option('ets_load_more_button'); 	
-			$pagingType = get_option('ets_product_qa_paging_type' ); 
-			$all_questions = get_post_meta( $productId,'ets_question_answer', true );
+		} 
+		 		
+	}
 
-			if($all_questions && is_array($all_questions)){
-				
-				$etsGetQuestion = array_filter($all_questions, function ($filterQuestion) {
-					return (isset($filterQuestion['approve']) && $filterQuestion['approve'] == 'yes') || !isset($filterQuestion['approve']);
-				});
-			}
+	/**
+ 	*Prepare QA listing data.
+ 	*/
+	public function display_qa_listing(){
+		global $product; 
+		$productId = $product->get_id(); 
```

Comment:
This method is now being used in the callback for QA listing shortcode as well, users may add the shortcode on other pages / posts as well intentionally / unintentionally.

if there is no global 1$product` variable (no set or null or blank) then this function should return blank string.